src/main/java/org/apache/accumulo/testing/continuous/ManySplits.java

```diff
@@ -0,0 +1,266 @@
+            while (true) {
+              Collection<Long> tabletFileSizes = getTabletFileSizes(client, tableName).values();
+              // filter out the tablets that are already the expected size
+              Set<Long> offendingTabletSizes =
+                  tabletFileSizes.stream().filter(tabletFileSize -> tabletFileSize > splitThreshold)
+                      .collect(Collectors.toSet());
+              // if all tablets are good, move on
+              if (offendingTabletSizes.isEmpty()) {
+                break;
+              }
+
+              elapsedMillis += sleepMillis;
+              // log every 3 seconds
+              if (elapsedMillis % SECONDS.toMillis(3) == 0) {
+                double averageFileSize =
+                    offendingTabletSizes.stream().mapToLong(l -> l).average().orElse(0);
+                long diff = (long) (averageFileSize - splitThreshold);
+                log.info(
+                    "{} tablets have file sizes not yet <= {} on table {}. Diff of avg offending file(s): {}",
+                    offendingTabletSizes.size(), splitThresholdStr, tableName,
+                    bytesToMemoryString(diff));
+              }
+              MILLISECONDS.sleep(sleepMillis);
+            }
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        });
+
+        long timeTaken = System.nanoTime() - beforeThresholdUpdate;
+
+        log.info(
+            "Time taken for all tables to reach expected total file size ({}): {} seconds ({}ms)",
+            splitThresholdStr, NANOSECONDS.toSeconds(timeTaken), NANOSECONDS.toMillis(timeTaken));
+      }
+
+      log.info("Test completed successfully.");
+
+      log.info("Deleting tables");
+      tableNames.stream().parallel().forEach(tableName -> {
+        try {
+          client.tableOperations().delete(tableName);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      });
+      log.info("Deleting namespace");
+      client.namespaceOperations().delete(NAMESPACE);
+
+    }
+  }
+
+  /**
+   * @return a map of tablets to the sum of their file size
+   */
+  private static Map<Text,Long> getTabletFileSizes(AccumuloClient client, String tableName)
+      throws TableNotFoundException, AccumuloException, AccumuloSecurityException {
+    TableId tableId = TableId.of(client.tableOperations().tableIdMap().get(tableName));
+    try (Scanner scanner = client.createScanner("accumulo.metadata")) {
+      scanner.fetchColumnFamily("file");
+      scanner.setRange(createMetaRangeForDefaultTablet(tableId.canonical()));
+
+      Map<Text,Long> result = new HashMap<>();
+      for (var entry : scanner) {
+        String encodedDFV = new String(entry.getValue().get(), UTF_8);
+        String[] ba = encodedDFV.split(",", 2);
+        long tabletFileSize = Long.parseLong(ba[0]);
+        result.merge(entry.getKey().getRow(), tabletFileSize, Long::sum);
```

Comment:
This is a neat way to sum up the per tablet sizes

Reply:
The String parsing was borrowed from the `DataFileValue` code :+1:. Had to re-implement it since its not public API